### ISSUE TITLE
a11y: Add Language attribute to page

### ISSUE THREAD

**Issue by derekjackson-das** [danny-avila/LibreChat#3458]
Language of the page missing
Identifying the language of the page or page elements allows screen readers to read the content in the appropriate language. It also facilitates automatic translation of content. | 
WCAG 3.1.1 Language of Page - A 
Identify the document language using the <html lang> attribute with a valid value (e.g., <html lang="en">). Ensure that all lang attribute values are valid.

**Comment by jacobcolyvan** [comment]
I looked into and Im unsure the best way to approach this.

My understanding of this is that we want the lang attribute to be set on the root html tag, eg:
`<html lang="en-AU">`.

This means we need to:
1. **set it in the initial markup**: ensure that screen readers can interpret it on page load.
2. **Update it dynamically**: When a user changes their language via the admin interface, it should update the tag as well.

Point 2 is easy enough, we just need to update `changeLang`, something like this:
```
  const changeLang = useCallback(
    (value: string) => {
      setSelectedLang(value);
      if (value === 'auto') {
        const userLang = navigator.language || navigator.languages[0];
        setLangcode(userLang);
        localStorage.setItem('lang', userLang);
        // Update HTML lang attribute
        document.documentElement.lang = userLang;
      } else {
        setLangcode(value);
        localStorage.setItem('lang', value);
        // Update HTML lang attribute
        document.documentElement.lang = value;
      }
    },
    [setLangcode, setSelectedLang],
  );
  ```

___

Point 1 is more complex. Since we're already saving the language preference in local storage, the simplest way is to set it on page load in the Root component, something like:
```
  useEffect(() => {
    const lang = localStorage.getItem('lang') || navigator.language || 'en-US';
    document.documentElement.lang = lang;
  }, []);
```
owever, this approach doesn't set the language attribute in the initial markup, which screen readers expect. However it does get us most of the way there without significant code changes.

I explored setting the lang attribute on the initial page load by:
1.  to add our local storage `lang` var onto every request from the frontend,
2. adding the lang tag into the initial markup payload based on the request from the frontend.

But, the initial `index.html` request won't include our intercepted lang attribute since the frontend hasn't loaded yet (to add it).

The most reliable way to achieve this would be to set a preferred-language cookie, which is sent with the initial request (currently it sets it in local storage). But this adds a little bit more complexity to the language process that I'm not sure we need.
I think if we go down this path we should replace the lang attribute in local storage with a cookie instead (across the codebase) so that the backend knows our preferred language with every request without us having to set it in the frontend.

I'm happy to take this approach but want to confirm if I'm missing something obvious or how we should proceed based on these points. @danny-avila (or anyone else), your opinions would be appreciated.

**Comment by danny-avila** [comment]
@jacobcolyvan sounds like a solid approach!

**Comment by danny-avila** [comment]
This PR "fixes" the linting error for the language attribute missing, but I believe @jacobcolyvan's approach would fully resolve the concern
